D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
-    - match: (\-\-|\+\+)
-      scope: keyword.operator.increment-decrement.d
-    - match: (\-|\+|\*|\/|~|%)
+      set: value
+    - match: '='
+      scope: keyword.operator.assignment.d
+      set: value
+    - match: '!'
+      scope: keyword.operator.d
+      set:
+        - match: '\('
+          scope: punctuation.section.parens.begin.d
+          set: [value-parens-after, value-list]
+        - match: '(?=\S)'
+          set: value
+    # This can also be a pointer in some situations
+    - match: \*
       scope: keyword.operator.arithmetic.d
-    - match: (!|&&|\|\|)
+      set:
+        - match: '(?=\)|,|;)'
+          pop: true
+        - match: '(?=\S)'
+          set: value
+    - match: '\?'
       scope: keyword.operator.logical.d
-    - match: \b(opNeg|opCom|opPostInc|opPostDec|opCast|opAdd|opSub|opSub_r|opMul|opDiv|opDiv_r|opMod|opMod_r|opAnd|opOr|opXor|opShl|opShl_r|opShr|opShr_r|opUShr|opUShr_r|opCat|opCat_r|opEquals|opEquals|opCmp|opCmp|opCmp|opCmp|opAddAssign|opSubAssign|opMulAssign|opDivAssign|opModAssign|opAndAssign|opOrAssign|opXorAssign|opShlAssign|opShrAssign|opUShrAssign|opCatAssign|opIndex|opIndexAssign|opCall|opSlice|opSliceAssign|opPos|opAdd_r|opMul_r|opAnd_r|opOr_r|opXor_r)\b
-      scope: keyword.operator.overload.d
-    - match: \b(new|delete|typeof|typeid|cast|align|is)\b
-      scope: keyword.other.d
-    - match: \b(new|throws)\b
-      scope: keyword.other.class-fns.d
-    - match: \b(package|extern)\b
-      scope: keyword.other.external.d
-    - match: \b(deprecated|unittest|debug)\b
-      scope: keyword.other.debug.d
-    - match: \b(u_char|u_short|u_int|u_long|ushort|uint|u_quad_t|quad_t|qaddr_t|caddr_t|daddr_t|dev_t|fixpt_t|blkcnt_t|blksize_t|gid_t|in_addr_t|in_port_t|ino_t|key_t|mode_t|nlink_t|id_t|pid_t|off_t|segsz_t|swblk_t|uid_t|id_t|clock_t|size_t|ssize_t|time_t|useconds_t|suseconds_t)\b
-      scope: support.type.sys-types.c
-    - match: \b(pthread_attr_t|pthread_cond_t|pthread_condattr_t|pthread_mutex_t|pthread_mutexattr_t|pthread_once_t|pthread_rwlock_t|pthread_rwlockattr_t|pthread_t|pthread_key_t)\b
-      scope: support.type.pthread.c
-    - match: \b(int8_t|int16_t|int32_t|int64_t|uint8_t|uint16_t|uint32_t|uint64_t|int_least8_t|int_least16_t|int_least32_t|int_least64_t|uint_least8_t|uint_least16_t|uint_least32_t|uint_least64_t|int_fast8_t|int_fast16_t|int_fast32_t|int_fast64_t|uint_fast8_t|uint_fast16_t|uint_fast32_t|uint_fast64_t|intptr_t|uintptr_t|intmax_t|intmax_t|uintmax_t|uintmax_t)\b
-      scope: support.type.stdint.c
-  all-types:
-    - include: support-type-built-ins-d
-    - include: support-type-d
-    - include: storage-type-d
-  comments:
-    - match: /\*
-      scope: punctuation.definition.comment.d
+      set: [value-condition-after, value]
+    - match: '(?=function|delegate)'
+      set: [value-after, basic-type2]
+    - match: '(?=\S)'
+      pop: true
+  value-condition-after:
+    - match: ':'
+      scope: keyword.operator.logical.d
```

Comment:
This seems like an odd scope for `:`, but I am not sure of how it is used

Reply:
This if for `cond ? v1 : v2`, same meaning as C++

Reply:
So in C we are currently using `keyword.operator.ternary`. `logical` is mostly used for `!`, `not`, `&&` and `||`.